org.eclipse.draw2d/src/org/eclipse/draw2d/graph/CompoundPopulateRanks.java

```diff
@@ -16,21 +16,22 @@
  * Places nodes into ranks for a compound directed graph. If a subgraph spans a
  * rank without any nodes which belong to that rank, a bridge node is inserted
  * to prevent nodes from violating the subgraph boundary.
- * 
+ *
  * @author Randy Hudson
  * @since 2.1.2
  */
 class CompoundPopulateRanks extends PopulateRanks {
 
+	@Override
 	public void visit(DirectedGraph g) {
 		CompoundDirectedGraph graph = (CompoundDirectedGraph) g;
 
 		/**
 		 * Remove long containment edges at this point so they don't affect MinCross.
 		 */
-		Iterator containment = graph.containment.iterator();
+		Iterator<Edge> containment = graph.containment.iterator();
 		while (containment.hasNext()) {
-			Edge e = (Edge) containment.next();
+			Edge e = containment.next();
 			if (e.getSlack() > 0) {
 				graph.removeEdge(e);
 				containment.remove();
```

Comment:
It'd also use a for-each loop here. Perhaps even something like `graph.containment.stream().filter(...).forEach(...)` to also get rid of the if() check.

Reply:
This should be part of a separate pull request. Will fix it there.

Reply:
I just noticed the `graph.removeEdge(e)` changes the container which is iterated over. Will the `stream().filter().foreEch()` work in this case?

Reply:
Good catch. I also missed the call to containment.remove(); In that case, the iterator has to be used.